lib/codegen/Decision.cpp

```diff
@@ -983,6 +996,15 @@ void makeStepFunction(
   initChoiceBuffer(
       dt, module, block, pre_stuck, fail, &choiceBuffer, &choiceDepth, &jump);
 
+  llvm::AllocaInst *HasSearchResults = nullptr;
+  if (search) {
+    HasSearchResults = new llvm::AllocaInst(
+        llvm::Type::getInt1Ty(module->getContext()), 0, "hasSearchResults",
+        block);
+    new llvm::StoreInst(
+        llvm::ConstantInt::getFalse(module->getContext()), HasSearchResults,
+        block);
+  }
```

Comment:
In the interests of keeping the backend documented moving forward, perhaps this code could include a comment explaining why we do this.